WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -196,17 +198,72 @@ - (void)viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
 
+    if (self.splitViewControllerIsHorizontallyCompact) {
+        [self animateDeselectionInteractively];
+    }
+
     [self.headerView setBlog:self.blog];
 
     // Configure and reload table data when appearing to ensure pending comment count is updated
     [self configureTableViewData];
-    [self.tableView reloadData];
+
+    [self reloadTableViewPreservingSelection];
     [self preloadBlogData];
 }
 
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
+{
+    [super traitCollectionDidChange:previousTraitCollection];
+
+    // Required to update disclosure indicators depending on split view status
+    [self reloadTableViewPreservingSelection];
+}
+
+- (void)showDetailViewForSubsection:(BlogDetailsSubsection)section
+{
+    switch (section) {
+        case BlogDetailsSubsectionStats:
+            [self.tableView selectRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:0]
+                                        animated:NO
+                                  scrollPosition:UITableViewScrollPositionNone];
+            [self showStats];
+            break;
+        case BlogDetailsSubsectionPosts:
+            [self.tableView selectRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:1]
+                                        animated:NO
+                                  scrollPosition:UITableViewScrollPositionNone];
+            [self showPostList];
+            break;
+        case BlogDetailsSubsectionThemes:
+        case BlogDetailsSubsectionCustomize:
+            if ([self.blog supports:BlogFeatureThemeBrowsing] || [self.blog supports:BlogFeatureMenus]) {
+                [self.tableView selectRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:2]
+                                            animated:NO
+                                      scrollPosition:UITableViewScrollPositionNone];
+                [self showThemes];
+            }
+            break;
+    }
+
+}
 
 #pragma mark - Data Model setup
 
+- (void)reloadTableViewPreservingSelection
+{
+    // First, we'll grab the appropriate index path so we can reselect it
+    // after reloading the table
+    NSIndexPath *selectedIndexPath = self.tableView.indexPathForSelectedRow ?: [NSIndexPath indexPathForRow:0 inSection:0];
```

Comment:
I think we talked about this one before, but should `selectedIndexPath` remain `nil` if `indexPathForSelectedRow` is? And then check for `nil` before selecting again?